infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutContextData.kt

```diff
@@ -80,14 +79,14 @@ sealed class LayoutContextData<T> : LayoutContextAware<T> {
 
         fun <T : LayoutAsset<T>> newDraft(branch: LayoutBranch): LayoutContextData<T> =
             when (branch) {
-                is MainBranch -> MainDraftContextData(null, null, null)
-                is DesignBranch -> DesignDraftContextData(null, null, null, branch.designId)
+                is MainBranch -> MainDraftContextData(null, null, null, null)
+                is DesignBranch -> DesignDraftContextData(null, null, null, null, branch.designId)
```

Comment:
🤔 Oheishuomiona tällainen nullilistan passaus näyttää aika merkilliseltä. Varmaan syystä näiden ContextData-luokkien kenttien arvot eivät ole default-null? (Saattaisi olla selvempää, että ainoastaan sen tietyn kentän (tai useamman) arvo joka oikeasti asetetaan olisi näissä vain näkyvissä)

Reply:
Onhan se vähän epätavallista. Näitä ei sinällään luoda monessa paikassa ja useampi noista kentistä on samaa tyyppiä (rowId) joten pidin default-arvoja vain turhana bugin mahdollisuutena jos jonnekin unohtuu lisätä kenttä. Noiden nullien asetuksella on myös selvä merkitys kun joissain siirtymissä (jotka on kaikki tässä tiedostossa) ne tarvitsee nimenomaa nullata ja toisissa ei -> halusin pitää tuon eksplisiittisenä. 

Kuitenkin, nullit peräkkäin ei ole kovin eksiplisiittistä eikä ainakaan selkeää. Voisin käyttää nimettyjä parametreja näissä, jolloin tuosta näkee selvemmin että mitä asetetaan. Muissa rakentajakutsuissa taitaakin jo olla, tästä vain uupuu.